src/metrics.go

```diff
@@ -17,35 +17,44 @@ import (
 	"github.com/newrelic/infra-integrations-sdk/log"
 )
 
-var metricsPlusDefinition = map[string][]interface{}{
-	"net.connectionsActive":            {"connections.active", metric.GAUGE},
-	"net.connectionsIdle":              {"connections.idle", metric.GAUGE},
-	"net.connectionsAcceptedPerSecond": {"connections.accepted", metric.RATE},
-	"net.connectionsDroppedPerSecond":  {"connections.dropped", metric.RATE},
-	"net.requestsPerSecond":            {"requests.total", metric.RATE},
-	"software.edition":                 {"edition", metric.ATTRIBUTE},
-	"software.version":                 {"version", metric.ATTRIBUTE},
-}
-
-var metricsPlusAPIDefinition = map[string]string{
-	"connections.active":   "net.connectionsActive",
-	"connections.idle":     "net.connectionsIdle",
-	"connections.accepted": "net.connectionsAcceptedPerSecond",
-	"connections.dropped":  "net.connectionsDroppedPerSecond",
-	"http.requests.total":  "net.requestsPerSecond",
-	"nginx.version":        "software.version",
-}
-
 var metricsStandardDefinition = map[string][]interface{}{
+	"nginx.edition":                    {"edition", metric.ATTRIBUTE},
+	"nginx.version":                    {"version", metric.ATTRIBUTE},
 	"net.connectionsActive":            {"active", metric.GAUGE},
 	"net.connectionsAcceptedPerSecond": {"accepted", metric.RATE},
 	"net.connectionsDroppedPerSecond":  {connectionsDropped, metric.RATE},
 	"net.connectionsReading":           {"reading", metric.GAUGE},
 	"net.connectionsWaiting":           {"waiting", metric.GAUGE},
 	"net.connectionsWriting":           {"writing", metric.GAUGE},
 	"net.requestsPerSecond":            {"requests", metric.RATE},
-	"software.edition":                 {"edition", metric.ATTRIBUTE},
-	"software.version":                 {"version", metric.ATTRIBUTE},
+}
+
+var metricsPlusDefinition = map[string][]interface{}{
+	"nginx.edition":                    {"edition", metric.ATTRIBUTE},
+	"nginx.version":                    {"version", metric.ATTRIBUTE},
+	"net.connectionsActive":            {"connections.active", metric.GAUGE},
+	"net.connectionsIdle":              {"connections.idle", metric.GAUGE},
+	"net.connectionsAcceptedPerSecond": {"connections.accepted", metric.RATE},
+	"net.connectionsDroppedPerSecond":  {"connections.dropped", metric.RATE},
```

Comment:
some fields seem common to nginx and nginx plus. Maybe we could have a metricsCommon with duplicated fields from both structs and embed the new structure on the standard and plus definitions.

Reply:
Yes, they are the same in the end, but the "source" is different and they way the integration is done is not easy to re-use so I just ignored it. There are 3 that could be extracted but I didn't think it was necessary for only those 3